core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1522,6 +1576,10 @@ public void trimUnreferencedReaders() throws IOException {
                 // we're shutdown potentially on some tragic event, don't delete anything
                 return;
             }
+            if (readers.isEmpty()) {
```

Comment:
maybe instead of adding this intermediate return we can invert and only do the work if it's not empty?